firefox-ios/Client/Frontend/Translations/TranslationsMiddleware.swift

```diff
@@ -106,26 +123,70 @@ final class TranslationsMiddleware {
 
     // TODO: FXIOS-13844 - Start translation a page and dispatch action after completion
     @MainActor
-    private func retrieveTranslations(for action: ToolbarMiddlewareAction) {
+    private func retrieveTranslations(for action: Action) {
         // We dispatch an action for now, but eventually we want to inject a script
         // to check if the page language differs from our locale language
         // When translation completed, we want icon to be active mode.
         Task { @MainActor in
-            await fetchData()
-            let toolbarAction = ToolbarAction(
-                translationConfiguration: TranslationConfiguration(
-                    prefs: profile.prefs,
-                    state: .active
-                ),
-                windowUUID: action.windowUUID,
-                actionType: ToolbarActionType.translationCompleted
-            )
-            store.dispatch(toolbarAction)
+            do {
+                try await fetchDataWithError()
+                dispatchAction(
+                    for: ToolbarActionType.translationCompleted,
+                    with: .active,
+                    and: action.windowUUID
+                )
+            } catch {
+                self.handleErrorFromTranslatingPage(for: action)
+            }
         }
     }
 
+    // When we receive an error translating the page, we want to update the translation
+    // icon on the toolbar to be inactive.
+    // We also want to display a toast.
+    private func handleErrorFromTranslatingPage(for action: Action) {
+        dispatchAction(
```

Comment:
```suggestion
        dispatchToolbarAction(
```

Maybe? I'm meh on this change